app/main.py

```diff
@@ -1,3 +1,16 @@
-def cache(func):
-    # Write your code here
-    pass
+from typing import Callable
+
+
+def cache(func: Callable) -> int:
+    cache_dict = {}
+
+    def wrapper(*args) -> int:
+        if str(args) not in cache_dict.keys():
+            print("Calculating new result")
+            result = func(*args)
+            cache_dict[str(args)] = result
```

Comment:
write inline this
and also do not use str(args), just args

Reply:
Because for example someone call with function with parameter=1 and will have another result if call with "1"